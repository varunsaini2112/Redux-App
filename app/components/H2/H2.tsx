import { ReactNode } from "react";
import { Text } from "react-native";

import { fontStyles } from "../../config/appStyles";

const H2 = ({ children }: { children: ReactNode }) => {
	return <Text style={fontStyles.h2}>{children}</Text>;
};

export default H2;
