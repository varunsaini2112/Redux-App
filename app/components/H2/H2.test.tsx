import { render, screen } from "@testing-library/react-native";
import H2 from "./H2";

describe("H2 Component", () => {
	it("should match snapshots", () => {
		const snapshot = render(<H2>Test Heading</H2>);

		expect(snapshot).toMatchSnapshot();
	});

	it("should render heading text", () => {
		render(<H2>Test Heading</H2>);

		expect(screen.getByText("Test Heading")).toBeOnTheScreen();
	});
});
