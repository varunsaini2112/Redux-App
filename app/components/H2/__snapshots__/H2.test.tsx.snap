// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`H2 Component should match snapshots 1`] = `
<Text
  style={
    {
      "color": "#132631",
      "fontSize": 20,
      "fontWeight": "600",
    }
  }
>
  Test Heading
</Text>
`;
